sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTest.cs

```diff
@@ -90,70 +91,52 @@ public void ThreadCGUD_MemberAUR_MessageGSU_NotificationT()
             ChatMessage message = chatThreadClient.GetMessage(messageId);
             #endregion Snippet:Azure_Communication_Chat_Tests_E2E_GetMessage
-            ChatMessage message2 = chatThreadClient2.GetMessage(messageId2);
-            ChatMessage message3 = chatThreadClient3.GetMessage(messageId3);
-            ChatMessage message4 = chatThreadClient3.GetMessage(messageId4);
-            ChatMessage message5 = chatThreadClient3.GetMessage(messageId5);
-            ChatMessage message6 = chatThreadClient3.GetMessage(messageId6);
+
+            var messagesMapping = new Dictionary<int, (string, ChatMessage)>
+            {
+                { 1, (message.Id, message) },
+                { 2, ExecuteSendGetMessage(2, chatThreadClient2, messageTemplate, displayNameTemplate, ChatMessagePriority.High) }
+            };
+            for (int i = 3; i < 7; i++)
+            {
+                messagesMapping.Add(i, ExecuteSendGetMessage(i, chatThreadClient3, messageTemplate, displayNameTemplate, ChatMessagePriority.High));
+            }
 
             #region Snippet:Azure_Communication_Chat_Tests_E2E_GetMessages
             Pageable<ChatMessage> messages = chatThreadClient.GetMessages();
             #endregion Snippet:Azure_Communication_Chat_Tests_E2E_GetMessages
-            Pageable<ChatMessage> messages2 = chatThreadClient2.GetMessages();
-            var getMessagesCount = messages.Count();
-            var getMessagesCount2 = messages2.Count();
+            Pageable<ChatMessage> messagesThread2 = chatThreadClient2.GetMessages();
+            var getMessagesCountThread1 = messages.Count();
+            var getMessagesCountThread2 = messagesThread2.Count();
 
             # region Pagination assertions
             Pageable<ChatMessage> messagesPaginationTest = chatThreadClient.GetMessages();
             string? continuationToken = null;
             var expectedPageSize = 2;
-            var messagesCounterTotal = 0;
+            var messagesTotal = 0;
             var messagesCounter = 0;
-            foreach (Page<ChatMessage> messagesPage in messagesPaginationTest.AsPages(continuationToken,2))
+            ulong previousId = ulong.MaxValue;
+            foreach (Page<ChatMessage> messagesPage in messagesPaginationTest.AsPages(continuationToken, expectedPageSize))
             {
-                messagesCounter = 0;
-                foreach (ChatMessage messagePage in messagesPage.Values)
-                {
-                    messagesCounterTotal++;
-                    messagesCounter++;
-                }
-                continuationToken = messagesPage.ContinuationToken;
-                //Last request does not return items
-                if (messagesPage.Values.Count == 0)
-                {
-                    Assert.IsNull(continuationToken);
-                }
-                else
+                (messagesCounter, continuationToken) = AssertPageableChatMessages(messagesPage, expectedPageSize, messagesMapping);
```

Comment:
What is messagesCounter? Is it the same as messagesPage .Values.Count?

Reply:
yeah, now that I will remove that function, no need to keep a separate counter.